broadinstitute/firecloud-orchestration
# Un-publish on Workspace Delete

## Addresses 
https://broadinstitute.atlassian.net/browse/GAWB-2352
Requires corresponding UI PR: https://github.com/broadinstitute/firecloud-ui/pull/931

## Changes:
Update the delete workspace logic to include un-publishing from Library if necessary.

## Testing:
* Delete a normal, un-published workspace and ensure no regressions
* Delete a published workspace and ensure that it does not show up in Library.
* Try to delete a published workspace with users that are not allowed.
  * Workspace "READER"
  * Workspace "OWNER" but not "CATALOG" permissions
  * "CATALOG" user, but not "OWNER"

---

I, the developer opening this PR, do solemnly pinky swear that:

- [x] I've followed [the instructions](../CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [x] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [x] I've updated the [FISMA documentation](../CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing

In all cases:

- [x] Squash and merge; you can delete your branch after this **unless it's for a hotfix**. In that case, don't delete it!
